### DISCUSSION TITLE
Local development setup guide

### DISCUSSION THREAD

**Discussion by mrks1337** [pkuehnel/TeslaSolarCharger#493] [Q&A] [answered]
Hi!
I would like to contribute to the frontend client and tried to get the project up and running locally but having problems with that.
At first I have to say that I am a macOS user, so I installed Visual Studio and cloned the repository.

When I start VS and run the project in it, my Chrome Browser is starting and showing Swagger with three routes but at this point I do not know how to get the frontend actually running locally.

If I'm able to get the client running my second question would be regarding test data. 
Is there something to do or to setup, so I get values shown in the application?

My personal goal would be to have it locally running so I could add Vue.js to the Razor Templates and work on UI/UX improvements (we quickly wrote in TFF forum) :)

Cheers

**Comment by pkuehnel** [comment]
> When I start VS and run the project in it, my Chrome Browser is starting and showing Swagger with three routes but at this point I do not know how to get the frontend actually running locally.

You need to set the correct startup project `TeslaSolarCharger.Server` and then you should get the normal frontend (if not, just remove the trailing `/swagger/index.html` from the URL).

> Is there something to do or to setup, so I get values shown in the application?

I just use my production TeslaMate as Test Data (I know it is not the best practice but as I need to test what really happens in the car, as Teslas API is not documented, this is the best way for me). To do that just go to your BaseConfiguration in your VS started TSC and update the URLs behind the dropdown:
![image](https://user-images.githubusercontent.com/35361981/210352946-73328a68-b43d-4c9a-88b5-4603b3773de9.png)

You just need to forward these ports in your `docker-compose.yml` and you are ready to go. The same with your Grid/Home Battery/Inverter Request URLs.

**Comment by mrks1337** [comment]
Hi! I tried it again (been some time...) and I get the following error in the client shown:

<img width="2953" alt="Screenshot 2023-06-16 at 12 47 02" src="https://github.com/pkuehnel/TeslaSolarCharger/assets/100295302/0366683b-b383-4903-af8f-a0fe220cf4d6">

Ideas what I am missing here? I am starting TeslaSolarCharger.Server and TeslaSolarCharger.Client simultaneously.

**Comment by pkuehnel** [comment, marked as answer]
You only need to start the server. The server references the client. So after starting the server, the client should be available (can't tell the Port, as this is randomly generated. For me it is `http://localhost:5021/`
